frontend/src/metabase/admin/databases/containers/DatabaseEditApp.unit.spec.tsx

```diff
@@ -112,6 +131,52 @@ describe("DatabaseEditApp", () => {
       expect(mockEvent.preventDefault).not.toHaveBeenCalled();
       expect(mockEvent.returnValue).toBe(undefined);
     });
+
+    it("does not show custom warning modal when leaving with no changes via SPA navigation", async () => {
+      const { history } = await setup({ initialRoute: "/home" });
+
+      history.push("/");
+
+      await waitForElementToBeRemoved(() =>
+        screen.queryAllByTestId("loading-spinner"),
+      );
+
+      const databaseForm = await screen.findByLabelText("Display name");
+      userEvent.type(databaseForm, "ab");
+      userEvent.type(databaseForm, "{backspace}{backspace}");
```

Comment:
👍 it's great that you test modifying the form back to the original state. Not just do nothing to the form.